### PR TITLE
api: send transactions ourselves

### PR DESCRIPTION
**Pull request by nalinbhardwaj**
Fixes potential nonce out of syncs

### CODE REVIEW

File: packages/daimo-api/src/network/viemClient.ts [hunk 1]
```diff
@@ -60,13 +62,16 @@ export class ViemClient {
   // Lock to ensure sequential nonce for walletClient writes
   private lockNonce = new AwaitLock();
   private nextNonce = 0;
+  public account: Account;
 
   constructor(
     private l1Client: PublicClient<Transport, Chain>,
     public publicClient: PublicClient<Transport, Chain>,
-    public walletClient: WalletClient<Transport, Chain, Account>,
+    private walletClient: WalletClient<Transport, Chain, Account>,
```
Comment (dcposch) [inline packages/daimo-api/src/network/viemClient.ts:70]: 👍 👍

File: packages/daimo-api/src/network/viemClient.ts [hunk 2]
```diff
@@ -122,31 +162,23 @@ export class ViemClient {
...
-        address: walletClient.account.address,
-        blockTag: "pending",
-      });
-      console.log(
-        `[CHAIN] exec ${args.functionName}, got tx count ${txCount}, updating nonce ${this.nextNonce}`
-      );
-      this.nextNonce = Math.max(this.nextNonce, txCount);
+    console.log(`[CHAIN] exec ${args.functionName}`);
+    const ret = await this.runWithOverrideParams(args, (args) =>
+      this.walletClient.writeContract(args)
```
Comment (dcposch) [inline packages/daimo-api/src/network/viemClient.ts:167]: 👍

File: packages/daimo-api/src/network/viemClient.ts [hunk 3]
```diff
@@ -109,6 +114,41 @@ export class ViemClient {
...
+      `[CHAIN] nonce: got tx count ${txCount}, updating nonce ${this.nextNonce}`
+    );
+    this.nextNonce = Math.max(this.nextNonce, txCount);
+  }
+
+  async runWithOverrideParams<
+    Args extends { nonce?: number; gas?: bigint },
+    Ret
+  >(args: Args, fn: (args: Args) => Ret): Promise<Ret> {
+    console.log(`[CHAIN] ready to run with override, waiting for lock`);
```
Comment (dcposch) [inline packages/daimo-api/src/network/viemClient.ts:132]: recommend logging timing:

```
const startMs = nowMs(), txId = Math.floor(Math.random()*1e6);
// ...
console.log(`[CHAIN] tx ${txId} ${nowMs() - startMs}ms: got lock`);
// ...
console.log(`[CHAIN] tx ${txId} ${nowMs() - startMs}ms: updated nonce ${this.nextNonce}`);
// ...
console.log(`[CHAIN] tx ${txId} ${nowMs() - startMs}ms: sent ${ret}`);
```

**Review by dcposch — APPROVED** [review]
LGreatTM

**Comment by nalinbhardwaj** [comment]
tested by sending a handful of different txes on testnet local

File: packages/daimo-api/src/network/viemClient.ts (hunk 3 above)
Comment (nalinbhardwaj) [reply packages/daimo-api/src/network/viemClient.ts]: done

**Pull request merged**